msgpack/_packer.pyx

```diff
@@ -274,7 +274,9 @@ cdef class Packer(object):
                 if ret == 0:
                     ret = msgpack_pack_raw_body(&self.pk, <char*>view.buf, L)
                 PyBuffer_Release(&view);
-            elif self.datetime and PyDateTime_CheckExact(o) and datetime_tzinfo(o) is not None:
+            elif self.datetime and PyDateTime_CheckExact(o):
+                if datetime_tzinfo(o) is None:
+                    PyErr_Format(ValueError, b"can not serialize '%.200s' object where tzinfo=None", Py_TYPE(o).tp_name)
```

Comment:
This is backward incompatible change.
User can not use `default` function to pack datetime without tzinfo.